src/main/java/com/yelp/nrtsearch/server/luceneserver/StartIndexHandler.java

```diff
@@ -65,6 +74,9 @@ public StartIndexResponse handle(IndexState indexState, StartIndexRequest startI
           } else {
             throw new IllegalStateException("Index " + indexState.name + " already restored");
           }
+        } catch (IOException e) {
+          logger.info("Unable to delete existing index data", e);
+          throw new StartIndexHandlerException(e);
         } finally {
           shardState.setRestored(true);
         }
```

Comment:
Do we still want this finally block to run if an exception is thrown?

Reply:
@aprudhomme I am actually not sure what purpose that line is serving. It essentially blocks an index from being restored until it is deleted and recreated if there is any error. Should we just remove it?
Also cc: @umeshdangat for any additional context here

Reply:
I don't have as much context on this code, but I would think you would want to remove the finally block and set restored true right after the artifact download above.

Reply:
Made the change, can revisit removing or modifying this check later